checkov/common/util/tqdm_utils.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import os
+import sys
+
+from colorama import Fore, Back
+from tqdm import tqdm  # type: ignore
+
+DEFAULT_BAR_FORMAT = '{l_bar}%s{bar:20}%s|[{n_fmt}/{total_fmt}]{postfix}' % (Fore.WHITE, Fore.RESET)
+SLOW_RUNNER_BAR_FORMAT = '{l_bar}%s{bar:20}%s|[{n_fmt}/{total_fmt}] %s[Slow Runner Warning]%s{postfix}' %\
+                         (Fore.LIGHTBLACK_EX, Fore.RESET, Back.YELLOW, Back.RESET)
+
+SLOW_RUNNERS = {'sca_package', 'terraform', 'cloudformation', 'helm', 'kubernetes', 'kustomize', 'secrets'}
+LOGS_ENABLED = os.environ.get('LOG_LEVEL', False)
+
+
+class ProgressBar:
+    def __init__(self, framework: str) -> None:
+        self.pbar = None
+        self.is_off = not self.should_show_progress_bar()
+        self.framework = framework
+
+    def initiate(self, total: int) -> None:
+        if total <= 0:
+            self.is_off = True
+
+        if self.is_off:
+            return
+
+        if self.pbar is not None:
+            self.pbar.reset(total)
+        else:
+            self.pbar = tqdm(total=total,
+                             bar_format=self.get_progress_bar_format(self.framework),
+                             desc=f'[ {self.framework} framework ]')
+
+    def update(self, value: int = 1) -> None:
+        if self.is_off:
+            return
+        self.pbar.update(value)  # type: ignore
```

Comment:
I think it would be good to check, `if not self.pbar` and then either log an error, that the progress bar was not correctly initiated or raise an Exception, then you can also remove the type ignore. And of course do the same thing for the other methods

Reply:
Decided to throw exceptions since if progress bars are used then LOG_LEVEL is unset and vice versa